### ISSUE TITLE
[Haskell] Unwanted `let` inserted when typeclass declaration contains typeclass constraint

### ISSUE THREAD

**Issue by ethagnawl** [jpalardy/vim-slime#136]
Sending the following to GHCi results in a parse error:

```
-- vim
instance Eq a => Eq (Identity a) where
  (==) (Identity v) (Identity v') = v == v'

-- GHCi
Prelude> :{
Prelude| let instance Eq a => Eq (Identity a) where
Prelude|       (==) (Identity v) (Identity v') = v == v'
Prelude| :}
<interactive>:0:0: parse error on input `instance'
```
For some reason, the typeclass constraint is causing the `let` to be introduced. A similar (albeit contrived) typeclass declaration _without_ a typeclass constraint works as expected:
```
-- vim 
instance Eq (Identity a) where
  (==) (Identity v) (Identity v') = False

-- GHCi
Prelude> :{
Prelude| instance Eq (Identity a) where
Prelude|       (==) (Identity v) (Identity v') = False
Prelude| :}
```

I'll take a pass it implementing a fix when I find a few minutes.

**Comment by jpalardy** [comment]
My own contribution to that file is minimal. Feel free to ping someone from the [blame](https://github.com/jpalardy/vim-slime/blame/master/ftplugin/haskell/slime.vim)
if needed.

**Comment by ethagnawl** [comment]
Closed via #137.
